### PR TITLE
[ASM] - XPSUPPORT - 2951 (EXPANDR) - Improved Fetching and Searchable Fields

### PR DESCRIPTION
**Pull request by BigEasyJ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
[XPSUPPORT-2951](https://jira-dc.paloaltonetworks.com/browse/XPSUPPORT-2951)

## Description

- Added support for looking back and paging for fetching Xpanse alerts. This should help assist customers that have a high throughput of alerts in XSOAR.
- Added a new command `asm-reset-last-run` to reset the fetch incidents commands last run value.
- Update incident fields to be searchable and usable for dashboard graphs.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , please review

**Review by johnnywilkes — APPROVED** [review]
This is approved

**Comment by johnnywilkes** [comment]
@rshunim , this has been approved by Xpanse.  Please merge when possible

**Comment by rshunim** [comment]
Waiting for @ShirleyDenkberg review and then will merge it

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 1]
```diff
@@ -91,12 +106,21 @@ configuration:
...
 - additionalinfo: Maximum amount of search results for listing commands.
   defaultvalue: '100'
   display: Search Limit
   name: search_limit
   required: false
   type: 0
+  section: Collect
+- defaultvalue: 0
+  display: 'Advanced: Minutes to look back when fetching (Warning: exceeding 720 could lead to issues with fetching)'
+  additionalinfo: Use this parameter to determine how long backward to look in the search for incidents that were created before the last run time and did not match the query when they were created.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:119]:
```suggestion
  additionalinfo: Use this parameter to determine how far back in time to look in the search for incidents that were created before the last run time and did not match the query when they were created.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 2]
```diff
@@ -7,20 +7,22 @@ This integration was integrated and tested with version 2.0 of Cortex Expander.
...
+| Alert Severities to Fetch                                                        | The severity of the alerts that will be fetched. If no severity is provided then alerts of all the severities will be fetched. Note: An alert whose status was changed to a filtered status after its creation time will not be fetched.                | False        |
+| Source Reliability                                                               | Reliability of the source providing the intelligence data. Used for !ip and !domain commands.                                                                                                                                                           | False        |
+| Look Back (Minutes to look back when fetching) | Use this parameter to determine how long backward to look in the search for incidents that were created before the last run time and did not match the query when they were created. | False        |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:24]:
```suggestion
| Look Back (Minutes to look back when fetching) | Use this parameter to determine how far back in time to look in the search for incidents that were created before the last run time and did not match the query when they were created. | False        |
```

File: Packs/CortexXpanse/ReleaseNotes/1_2_3.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+Allowed the field to be searchable and usable for dashboard graphs.
+
+##### Xpanse Progress Status
+
+Allowed the field to be searchable and usable for dashboard graphs.
+
+##### Xpanse Port
+
+Allowed the field to be searchable and usable for dashboard graphs.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/ReleaseNotes/1_2_3.md:94]:
```suggestion
Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Provider

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Alert ID

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Service Validation

Allows  the field to be searchable and usable for dashboard graphs.

##### Xpanse Protocol

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Category

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Cloud Asset

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Service Classifications

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse External ID

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Host Name

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Responsive IP Asset

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Country Code

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Service ID

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Business Units

Allows  the field to be searchable and usable for dashboard graphs.

##### Xpanse Certificate Asset

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse IP

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Service Details

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Attack Surface Rule Name

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Asset IDs

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Domain Asset

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Description

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Progress Status

Allows the field to be searchable and usable for dashboard graphs.

##### Xpanse Port

Allows the field to be searchable and usable for dashboard graphs.
```

File: Packs/CortexXpanse/ReleaseNotes/1_2_3.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+
+#### Integrations
+
+##### Cortex Xpanse
+
+- Added support for looking back and paging for fetching Xpanse alerts. This should help assist customers that have a high throughput of alerts in XSOAR.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/ReleaseNotes/1_2_3.md:101]:
```suggestion
- Added support for looking back and paging for fetching Cortex Xpanse alerts. This helps customers who have a high throughput of alerts in Cortex XSOAR.
```

**Comment by ShirleyDenkberg** [comment]
@johnnywilkes @efelmandar @rshunim Doc review completed.

**Comment by rshunim** [comment]
Hi @BigEasyJ,
@ShirleyDenkberg approved the docs.
Please fix her comments.
Thanks

**Comment by johnnywilkes** [comment]
@rshunim , incorporated @ShirleyDenkberg 's comments (thank you!), please merge when possible

**Review by rshunim — APPROVED** [review]
Great Job!

**Pull request merged**